### PR TITLE
Do not remove parentheses when it would cause a parsing change in a 'when clause'

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/75203

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal :-)

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs [hunk 1]
```diff
@@ -259,7 +259,17 @@ nodeParent is BinaryExpressionSyntax(SyntaxKind.CoalesceExpression) binary &&
 
         // case x when (y): -> case x when y:
         if (nodeParent.IsKind(SyntaxKind.WhenClause))
+        {
+            // Subtle case, `when (x?[] ...):`.  Can't remove the parentheses here as it can the conditional access
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs:263]: >can [](http://example.com/codeflow?start=89&length=3)

nit: typo?

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs [hunk 2]
```diff
@@ -259,7 +259,17 @@ nodeParent is BinaryExpressionSyntax(SyntaxKind.CoalesceExpression) binary &&
...
+            // become a conditional expression.
+            for (var current = expression; current != null; current = current.ChildNodes().FirstOrDefault() as ExpressionSyntax)
```
Comment (ToddGrun) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs:265]: >ChildNodes().FirstOrDefault() [](http://example.com/codeflow?start=78&length=29)

I'm a little surprised we don't have a FirstChildNode

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs:263]: will fix in followup.

**Pull request merged**